src/components/styled-numberic-input/styled-numeric-input.tsx

```diff
@@ -0,0 +1,61 @@
+import { inRange } from 'lodash';
+import React, { useCallback } from 'react';
+import { TextProps } from 'react-native';
+
+import { StyledTextInput } from '../styled-text-input/styled-text-input';
+
+type Props = TextProps & {
+  min?: number;
+  max?: number;
+  value: number | string;
+  multiline?: boolean;
+  onChange: (v: number | string) => void;
```

Comment:
`value` and `onChange` should only accept/return `number`. Assertion to/from `string` must be inside this component only.  
Remove `multiline `, there is no case right now when the number input is going to be multiline.  
Remove `TextProps` and avoid prop drilling.